openedx/core/djangoapps/user_authn/views/tests/test_reset_password.py

```diff
@@ -123,21 +123,23 @@ def test_nonexist_email_password_reset(self):
         'openedx.core.djangoapps.user_authn.views.password_reset.render_to_string',
         Mock(side_effect=mock_render_to_string, autospec=True)
     )
-    def test_password_reset_ratelimited(self):
+    @ddt.data(True, False)
+    def test_password_reset_ratelimited(self, existing_user):
         """
-        Test that reset password endpoint only allow one request per minute.
+        Test that reset password endpoint only allow one request per minute for both
+        existing and non-existing users.
         """
         cache.clear()
-
-        password_reset_req = self.request_factory.post('/password_reset/', {'email': 'thisdoesnotexist@foo.com'})
-        password_reset_req.user = AnonymousUser()
+        email = self.user.email if existing_user else 'thisdoesnotexist@foo.com'
+        password_reset_req = self.request_factory.post('/password_reset/', {'email': email})
+        password_reset_req.user = self.user if existing_user else AnonymousUser()
+        password_reset_req.site = Mock(domain='example.com')
         good_resp = password_reset(password_reset_req)
         self.assertEqual(good_resp.status_code, 200)
```

Comment:
Can we check if the `password_reset_email_limiter.tick_request_counter(request)` have triggered ?

Reply:
@awaisdar001 I believe asserting 403 is sufficient to verify that, isn't it?

Reply:
As we are changing where the counter will tick, let's test if it has actually triggered here.